coderedcms/models/page_models.py

```diff
@@ -771,16 +771,23 @@ def most_recent_occurrence(self):
         """
         Gets the next upcoming, or last occurrence if the event has no more occurrences.
         """
-        noc = self.next_occurrence()
-        if noc:
-            return noc
-        else:
+
+        try:
+            noc = self.next_occurrence()
+            if noc:
+                return noc
             aoc = []
             for occurrence in self.occurrences.all():
                 aoc += [instance for instance in occurrence.all_occurrences()]
             if len(aoc) > 0:
                 return aoc[-1] # last one in the list
-            return None
+
+        except AttributeError:
+            # Triggers when a preview is initiated on an EventPage because it uses a FakeQuerySet object.
+            # Here we manually compute the next_occurrence
+            occurrences = [e.next_occurrence() for e in self.occurrences.all()]
```

Comment:
The code is very clear. I'm just really confused why this works:
```
occurrences = [e.next_occurrence() for e in self.occurrences.all()]
```
but this doesn't:
```
aoc += [instance for instance in occurrence.all_occurrences()]
```

Reply:
That's not the part that breaks.  What breaks is calling `self.next_occurrence()` because that calls related fields that the fake queryset doesn't have access to.